### PR TITLE
Move DisableResyncOnSDKServer to Beta

### LINKED ISSUE #3731: Graduate DisableResyncOnSDKServer to Beta
Is your feature request related to a problem? Please describe.
Graduate "DisableResyncOnSDKServer" feature to beta and enable it by default.

Describe the solution you'd like
Move to Beta to get some more testing on the feature before moving it to Stable.

Describe alternatives you've considered
Remove the functionality.

Additional context
Task list:

https://github.com/googleforgames/agones/blob/065052a9beead20bbc2d7bf9a18a7906bc73a882/pkg/util/runtime/features.go#L85-L91

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3731 

**Special notes for your reviewer**:

### CODE REVIEW

File: install/helm/agones/defaultfeaturegates.yaml [hunk 1]
```diff
@@ -16,10 +16,11 @@
 
 # Beta features
 FleetAllocationOverflow: true
+DisableResyncOnSDKServer: true
```
Comment (zmerlynn) [inline install/helm/agones/defaultfeaturegates.yaml:19]: nit: keep sorted within `Beta`

File: site/content/en/docs/Guides/feature-stages.md [hunk 2]
```diff
@@ -40,8 +40,8 @@ The current set of `alpha` and `beta` feature gates:
 | Feature Name                                                                                                                | Gate                               | Default  | Stage   | Since  |
 |-----------------------------------------------------------------------------------------------------------------------------|------------------------------------|----------|---------|--------|
 | [Allocated GameServers are notified on relevant Fleet Updates][fleet-updates]                                               | `FleetAllocationOverflow`          | Enabled  | `Beta`  | 1.37.0 |
+| [DisableResyncOnSDKServer](https://github.com/googleforgames/agones/issues/3377)                                            | `DisableResyncOnSDKServer`         | Enabled  | `Beta`  | 1.40.0 |
```
Comment (zmerlynn) [inline site/content/en/docs/Guides/feature-stages.md:43]: nit: Keep sorted within `Beta` (so put above `FleetAllocationOverflow`).

File: pkg/util/runtime/features.go [hunk 3]
```diff
@@ -99,11 +99,11 @@ var (
 	// In each of these, keep the feature sorted by descending maturity then alphabetical
 	featureDefaults = map[Feature]bool{
 		// Beta features
+		FeatureDisableResyncOnSDKServer:         true,
 		FeatureFleetAllocateOverflow: true,
```
Comment (zmerlynn) [inline pkg/util/runtime/features.go:103]: huh, did this not fail `goimports`?

**Review by zmerlynn — CHANGES_REQUESTED** [review]

File: pkg/util/runtime/features.go (hunk 3 above)
Comment (Kalaiselvi84) [reply pkg/util/runtime/features.go]: used the `goimports` to format this file

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**